test/api/unit/top-level/dataexport.test.js

```diff
@@ -0,0 +1,79 @@
+import dataexport from '../../../../website/server/controllers/top-level/dataexport';
```

Comment:
thanks for the fix @benkelaar ! The only thing is, this test is about a controller so we should either:

1. Transform it into an integration test
2. Move the logic for the XML export outside of the controller into a library file (website/server/libs) and unit test that

Reply:
I think 2 is a good idea that I might make some time for.

What's the reason behind 1?

Reply:
The idea is that for the results of controllers, which is the result of making an HTTP call to them we use integration tests that actually do an HTTP call behind the scenes and check its results. Unit tests instead are used to test the behavior of functions calls. Does it make sense?

So in general:

- controllers -> get tested with integration tests
- anything else -> unit tests

So if in this case the behavior is difficult to test with an integration test we should move the code to the `libs` folder and make an unit test against a function that simply generates the XML, not that calls the XML generator and then checks that the response sent to the HTTP request is correct

Reply:
Cool, I understand.  Is this strategy documented somewhere?  Might be a good idea to add it to the CONTRIBUTING.md or the wiki.

Reply:
@paglias So I've been looking at where to put the integration test.  At first I wanted to add a new test run target `test:api-top-level:integration` (That's why I got to slightly refactoring `gulp-tests.js`), but now I notice that the payment endpoints are simply tested under `api-v3`.

Given the division of run targets I think it would be cleaner to add a `test:api-top-level:integration` target and have that run also the payment tests, but that would also mean setting up an extra github action to run those, and I don't think I have the rights to do that.

So shall I just add the integration test to `api-v3` or shall I create a new `api-top-level` test run target?

Reply:
Good point, let's add it to the api-v3 for now, apart from the naming which is indeed confusing it's not harmful

Reply:
Added an integration test and extracted the functionality in a library.